spec/facility_spec.rb

```diff
@@ -24,3 +24,237 @@
+      expect(@facility_1.registered_vehicles).to eq([])
+      expect(@facility_1.collected_fees).to eq(0)
+    end
+
+    it 'can register a new vehicle' do
+      @facility_1.add_service('Vehicle Registration')
+      @facility_1.register_vehicle(@cruz)
+      expect(@facility_1.registered_vehicles).to eq([@cruz])
+    end
+    
+    it 'should still have a nil registration date' do
+      expect(@cruz.registration_date).to eq nil
+    end
+
+    it 'should now have a registration date & a plate type' do
+      @facility_1.add_service('Vehicle Registration')
+      @facility_1.register_vehicle(@cruz)
+      expect(@cruz.date_registered).to eq(Date.today)
+      expect(@cruz.plate_finder).to eq(:regular)
+    end
+
+    it 'can total fees collected from registration' do
+      @facility_1.add_service('Vehicle Registration')
+      @facility_1.register_vehicle(@cruz)
+      expect(@facility_1.collected_fees).to eq(100)
+    end
+
+    it 'can register more vehicles' do
+      @facility_1.add_service('Vehicle Registration')
+      @facility_1.register_vehicle(@cruz)
+
+      # Here comes the 69 Camaro *vroom vroom*
+      @facility_1.register_vehicle(@camaro)
+      expect(@camaro.registration_date).to eq(Date.today)
+
+      # Here comes the EV Bolt *bzzZzzz*
+      @facility_1.register_vehicle(@bolt)
+      expect(@bolt.registration_date).to eq(Date.today)
+      expect(@bolt.plate_finder).to eq(:ev)
+
+      # There's 3 cars registered
+      expect(@facility_1.registered_vehicles).to eq([@cruz, @camaro, @bolt])
+
+      # The fees total after 3 vehicles have been registered
+      expect(@facility_1.collected_fees).to eq(325)
+    end
+
+    it 'can create a new facility with empty attributes' do
+      expect(@facility_2.registered_vehicles).to eq([])
+      expect(@facility_2.services).to eq([])
+      @facility_2.register_vehicle(@bolt)
+      expect(@facility_2.registered_vehicles).to eq([])
+      expect(@facility_2.collected_fees).to eq(0)
+    end
+  end
+end
+
+
+# ---------- GETTING A DRIVERS LICENSE ----------
+
+RSpec.describe Facility do
+  before(:each) do
+    @registrant_1 = Registrant.new('Bruce', 18, true )
+    @registrant_2 = Registrant.new('Penny', 16 )
+    @registrant_3 = Registrant.new('Tucker', 15 )
+    @facility_1 = Facility.new({name: 'Albany DMV Office', address: '2242 Santiam Hwy SE Albany OR 97321', phone: '541-967-2014' })
+    @facility_2 = Facility.new({name: 'Ashland DMV Office', address: '600 Tolman Creek Rd Ashland OR 97520', phone: '541-776-6092' })
+  end
+  describe "Iteration 2" do
+    it 'Registrants and Facilities exist' do
+      expect(@registrant_1).to be_an_instance_of(Registrant) 
+      expect(@registrant_2).to be_an_instance_of(Registrant) 
+      expect(@registrant_3).to be_an_instance_of(Registrant)
+      expect(@facility_1).to be_an_instance_of(Facility)
+      expect(@facility_2).to be_an_instance_of(Facility)
```

Comment:
Not necessary to test the Registrants in this class since you already do that in the Registrant class. Also, really only need to test one instance of Facility for this test.